host/src/lib/HelperFunctions.jsx

```diff
@@ -0,0 +1,114 @@
+import { isNullOrUndefined, GameSessionState } from "@righton/networking";
+/* 
+* counts all answers for current question using isChosen, for use in footer progress bar
+* @param {array} answerArray - array of answers for current question
+* @returns {number} count - number of answers for current question
+*/
+export const getTotalAnswers = (answerArray) => {
+  let count = 0;
+  if (answerArray) {
+    answerArray.forEach(answerCount => {
+      count = count + answerCount;
+    });
+    return count;
+  }
+  return count;
+};
+
+
+/*
+* returns the choices object for an individual question
+* @param {array} questions - array of questions
+* @param {number} currentQuestionIndex - index of current question
+* @returns {array} choices - array of choices for current question
+*/
+export const getQuestionChoices = (questions, currentQuestionIndex) => {
+  if (isNullOrUndefined(questions) || questions.length <= currentQuestionIndex || isNullOrUndefined(questions[currentQuestionIndex].choices)) {
+    return null;
+  }
+  return questions[currentQuestionIndex].choices;
+};
+
+
+/*
+* returns an array of which team names picked which question choices
+* @param {array} teamsArray - array of teams
+* @param {number} currentQuestionIndex - index of current question
+* @param {array} choices - array of choices for current question
+* @returns {array} teamsPickedChoices - array of objects containing corresponding team names and choice texts
+*/
+export const getTeamByQuestion = (teamsArray, currentQuestionIndex, choices, questions, currentState) => {
+  const teamsPickedChoices = [];
+
+  teamsArray.forEach(team => {
+    const teamPickedChoices = [];
+    let isNoResponse = true;
+    
+    team.teamMembers && team.teamMembers.forEach(teamMember => {
+      teamMember.answers && teamMember.answers.forEach(answer => {
+        if (answer.questionId === questions[currentQuestionIndex].id) {
+          if (((currentState === GameSessionState.CHOOSE_CORRECT_ANSWER || currentState === GameSessionState.PHASE_1_DISCUSS) && answer.isChosen) || ((currentState === GameSessionState.PHASE_2_DISCUSS || currentState === GameSessionState.CHOOSE_TRICKIEST_ANSWER) && answer.isTrickAnswer)) {
```

Comment:
Usually when the condition is too long, it's worth to break it down to multiple lines for readability:

```typescript
(
    (
        (
            currentState === GameSessionState.CHOOSE_CORRECT_ANSWER ||
            currentState === GameSessionState.PHASE_1_DISCUSS
        ) && answer.isChosen
    ) || (
        (
            currentState === GameSessionState.PHASE_2_DISCUSS ||
            currentState === GameSessionState.CHOOSE_TRICKIEST_ANSWER
    ) && answer.isTrickAnswer)
)
```

In cases like this that the conditions are a bit complex also, you can extract them into a variable, one for the first batch and one for the second batch

Reply:
Good call, fixed